src/ucp/proto/proto_perf.c

```diff
@@ -819,3 +826,82 @@ void ucp_proto_flat_perf_destroy(ucp_proto_flat_perf_t *flat_perf)
     ucs_array_cleanup_dynamic(flat_perf);
     ucs_free(flat_perf);
 }
+
+/**
+ * @brief Estimate the time for a UCP operation.
+ *
+ * This function selects the best protocol for the given operation parameters
+ * and message length, then estimates the time it would take.
+ *
+ * Memory information (type and system device) is determined based on available
+ * parameters with the following fallback logic:
+ * 1. If @a memh (memory handle) is provided, its properties are used.
+ * 2. Else, if @a rkey (remote key) is provided, its properties are used.
+ * 3. Else (both @a memh and @a rkey are NULL), host memory and an unknown
+ *    system device (@ref UCS_SYS_DEVICE_ID_UNKNOWN) are assumed.
+ *
+ * @param [in]  ep           Endpoint to perform the operation on.
+ * @param [in]  op_id        Operation ID (e.g., put, get, amo).
+ * @param [in]  memh         Optional UCP memory handle for the local data buffer.
+ *                           Can be NULL.
+ * @param [in]  rkey         Optional UCP remote key for the remote data buffer.
+ *                           Can be NULL.
+ * @param [in]  msg_length   Message length for the operation.
+ * @param [out] time_est     Filled with the estimated time in seconds.
+ *
+ * @return UCS_OK if successful, or an error code otherwise.
+ */
+ucs_status_t
+ucp_proto_est_perf(ucp_ep_h ep, ucp_operation_id_t op_id, ucp_mem_h memh,
```

Comment:
Consider `est` -> `estimate`.
First, usage of such abbreviations is generally discouraged, for ambiguity reason.
Second, consistency with other `_estimate_` funcs we have.